src/output-json.c

```diff
@@ -798,15 +796,23 @@ static int MacSetIterateToJSON(uint8_t *val, MacSetSide side, void *data)
     return 0;
 }
 
-int CreateJSONEther(JsonBuilder *js, const Packet *p, const MacSet *ms)
+int CreateJSONEther(JsonBuilder *js, const Packet *p, const Flow *f)
 {
-    jb_open_object(js, "ether");
     if (unlikely(js == NULL))
```

Comment:
why not check for `f` here as well? avoids having to do the cleanup from line 808

Reply:
Because it can happen that `f` is NULL -- this was the reason for the bug altogether. If `f` is NULL, then we want to use only packet metadata (lines 806-833). Line 808 is only there to make sure that we don't crash again in case `CreateJSONEther()` is called without neither `p` nor `f` being valid (I have no idea if that is possible). It should never happen.

If we check `f` as early you suggest, we will not be able to handle the packet case. Or am I not understanding correctly what your suggestion is?

Reply:
I missed the packet case. Merged as is, thanks!